thirdeye-server/src/main/java/ai/startree/thirdeye/resources/AlertResource.java

```diff
@@ -213,6 +230,8 @@ public Response validateMultiple(
       final AlertDTO existing =
           api.getId() == null ? null : ensureExists(dtoManager.findById(api.getId()));
       validate(api, existing);
+      ensureHasAccess(principal, api, AccessType.UPDATE);
+      ensureHasAccessToRelatedResources(principal, api);
```

Comment:
Not sure I understand this? So user should have access to alert template when getting alert?

Reply:
The idea is that a user can't create an alert if they don't have access to the alert template. Similarly, for dataset+datasource. But I'm not sure if that makes sense or if this is the best way to implement it.

I think I'll hold off on the related resources business for now.